tests/cli/test_mila_init_ssh_config.py

```diff
@@ -0,0 +1,349 @@
+from __future__ import annotations
+
+import contextlib
+import inspect
+import itertools
+import textwrap
+from contextlib import contextmanager
+from pathlib import Path
+
+import pytest
+from prompt_toolkit import PromptSession
+from prompt_toolkit.input.defaults import create_pipe_input
+
+from milatools.cli.commands import setup_ssh_config_interactive
+from milatools.cli.utils import SSHConfig
+
+expected_block_mila = """
+Host mila
+  HostName login.server.mila.quebec
+  User {user}
+  PreferredAuthentications publickey,keyboard-interactive
+  Port 2222
+  ServerAliveInterval 120
+  ServerAliveCountMax 5
+"""
+
+expected_block_mila_cpu = """
+Host mila-cpu
+  User {user}
+  Port 2222
+  ForwardAgent yes
+  StrictHostKeyChecking no
+  LogLevel ERROR
+  UserKnownHostsFile /dev/null
+  RequestTTY force
+  ConnectTimeout 600
+  ProxyCommand ssh mila "salloc --partition=unkillable --dependency=singleton --cpus-per-task=2 \
+--mem=16G /usr/bin/env bash -c 'nc \\$SLURM_NODELIST 22'"
+  RemoteCommand srun --cpus-per-task=2 --mem=16G --pty /usr/bin/env bash -l
+"""
+
+expected_block_mila_gpu = """
+Host mila-gpu
+  User {user}
+  Port 2222
+  ForwardAgent yes
+  StrictHostKeyChecking no
+  LogLevel ERROR
+  UserKnownHostsFile /dev/null
+  RequestTTY force
+  ConnectTimeout 600
+  ProxyCommand ssh mila "salloc --partition=unkillable --dependency=singleton --cpus-per-task=2 \
+--mem=16G --gres=gpu:1 /usr/bin/env bash -c 'nc \\$SLURM_NODELIST 22'"
+  RemoteCommand srun --cpus-per-task=2 --mem=16G --gres=gpu:1 --pty /usr/bin/env bash -l
+"""
+
+
+expected_block_compute_node = """
+Host *.server.mila.quebec !*login.server.mila.quebec
+  HostName %h
+  User {user}
+  ProxyJump mila
+"""
+
+
+def _join_blocks(*blocks: str, user: str = "bob") -> str:
+    return "\n".join(textwrap.dedent(block) for block in blocks).format(user=user)
+
+
+def _yn(accept: bool):
+    return "y" if accept else "n"
+
+
+def test_creates_ssh_config_file(tmp_path: Path):
+    ssh_config_path = tmp_path / "ssh_config"
+    with set_user_inputs(["y", "bob\r", "y", "y", "y", "y", "y"]) as input_pipe:
+        setup_ssh_config_interactive(tmp_path / "ssh_config", _input=input_pipe)
+    assert ssh_config_path.exists()
```

Comment:
```
    with set_user_inputs(["y", "bob\r", "y", "y", "y", "y", "y"]) as input_pipe:
        with unittest.mock.patch("questionary.confirm", new=functools.partial(questionary.confirm, input=input_pipe)), \
        with unittest.mock.patch("questionary.text", new=functools.partial(questionary.text, input=input_pipe)):
            setup_ssh_config_interactive(tmp_path / "ssh_config")
```

?

Reply:
```
def test_creates_ssh_config_file(tmp_path: Path, file_regression):
    [...]
    with open("ssh_config_path") as _f:
        file_regression.check(_f.read())
```

with `pytest-regression`?

Reply:
Fixed in 5647ea0 for the first part, and 14f005c + 2b6d6cb for file_regression